docs/architecture.md

```diff
@@ -112,76 +112,206 @@ The redux store is composed of two `slices` : `documents` and `queries`:
-`ObservableQuery` are objects instantiated by a <Query /> component. Their
-role is to react to store changes and wake the component. They should not
-be used directly as `useQuery` and `queryConnect` do the job for you.
+ℹ️ See the [react integration](/docs/react-integration.md) for more insight about the glue between the redux store and the UI.
 
 What do we mean exactly by saying "This redux store
 brings observability to cozy-client, and allows for connection of
 UI components to the data." Let's take a full exemple: 
 
-We have a component that displays a todolist: 
+Here is a simple React example on how to display data with cozy-client:
 ```jsx
-// TodoListComponent.jsx
 const { data, fetchStatus } = useQuery(Q('io.cozy.todos'), {'as': 'todoslist'})
+if(fetchStatus === 'loading'){
+  return <Spinner /> 
+}
 if(fetchStatus === 'loaded'){
   return <TodoLists todos={data}> /> 
 }
 ```
 
-But we also have a component that gives use the opportunity to add a 
-new todo: 
-```jsx
-client.save({_type: 'io.cozy.todos', 'label': 'New TODO'});
+This way, a spinner will be displayed during the time the query is actually run. Once the query is done, data is available and can be displayed.
+
+Any data change will be automatically handled by cozy-client.
+
+
+### Queries
+
+Here, we describe what happens internally when a query is called from an app.
+
+First, it is important to understand that each query is stored in the store, in a `queries` associative array. Each query has a unique id, a set of metadata, and the list of retrieved document ids.
+
+For example:
+```js
+{
+  query1: {
+    definition: {
+      ...
+    },
+    fetchStatus: "loaded",
+    lastFetch: 1716989816939,
+    lastUpdate: 1716989816939,
+    hasMore: false,
+    count: 2,
+    bookmark: "xyz",
+    data: ["docId1", "docId2"] // confusingly named `data`, but only stores ids
+  } 
+}
 ```
 
-After the call to `save` your `TodoListComponent` will be re-rendered 
-with the `New TODO`. 
+#### Fetch status
+
+An important attribute used for the query lifecycle is the `fetchStatus`.
+It can take the following values:
+- `pending`: the query is about to be run for the first time.
+- `loading`: the query is currently running and might return results. From a UX perspective, it is often used by apps to display a spinner.
+- `loaded`: the query had been run and returned results.
+- `failed`: the last query execution failed.
+
+When a query is called for the first time, it is first initialized in the `queries` collection in the store, with the `pending` status.
 
-### How does it works?
+If the query already exists in the store, its status is checked to ensure that it is not already in a `loading` state.
 
-When a `query` is resolved, CozyClient dispatches a `receiveQueryResult`
-action for a simple `get` but `receiveMutationResult` when we mutate 
-something. 
+Then, the query is "loaded", making its status on `loading`.
```

Comment:
on loaded?

Reply:
At this point, we dispatch the `loadQuery` (what I mean by "loaded"), setting the status from "loaded" to "loading" here: https://github.com/cozy/cozy-client/blob/master/packages/cozy-client/src/store/queries.js#L158-L160
But I agree that the loaded/loading terminology is a bit confusing, I remember that we had some brain damage here :smile: 
Maybe it would be more understandable to remove the "loaded" notion here and say something like this instead? 

>  Then, the query status is set on `loading` and the query actually run [...]

Reply:
I like what you suggest.